completions/hash

```diff
@@ -0,0 +1,26 @@
+# hash completion                                           -*- shell-script -*-
+
+_comp_cmd_hash()
+{
+    local cur prev words cword comp_args
+    _comp_initialize -- "$@" || return
+
+    local noargopts='!(*[p]*)'
+    # shellcheck disable=SC2254
+    case $prev in
+        -${noargopts}p)
+            _comp_compgen_filedir
+            return
+            ;;
+    esac
+
+    if [[ $cur == -* ]]; then
+        _comp_compgen_help
```

Comment:
bash < 4.4 doesn't support `hash --help`, so we need to use `help hash`.

```suggestion
        _comp_compgen_help -c help "$1"
```

Reply:
Thanks! Better that way in the first place.